version: "3.4"

services:
  consume:
    build:
      context: .
      target: test
      dockerfile: consumer/Dockerfile
    depends_on:
      - db
      - migrate
    env_file:
      - .env
    networks:
      - default
    command: ["python", "consumer_run.py"]
    restart: on-failure:5
    volumes:
      - .:/app

  produce:
    build:
      context: .
      target: test
      dockerfile: producer/Dockerfile
    env_file:
      - .env
    networks:
      - default
    command: ["python", "producer_run.py"]
    volumes:
      - .:/app

  migrate:
    build:
      context: .
      target: test
      dockerfile: consumer/Dockerfile
    env_file:
      - .env
    command: python consumer/database/migrate.py
    restart: on-failure:5
    depends_on:
      - db

  db:
    environment:
      POSTGRES_USER: aiven
      POSTGRES_PASSWORD: aiven
      POSTGRES_DB: aiven_default
    image: postgres:11.6-alpine
    networks:
      - default
    ports:
      - 5432:5432
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U aiven -d aiven_default"]
      interval: 10s
      timeout: 5s
      retries: 5
